### PR TITLE
Move project list to draughtsman

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/10965
Corresponding installations PR: https://github.com/giantswarm/installations/pull/1250

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -135,6 +135,7 @@ func mainError() error {
 	daemonCommand := newCommand.DaemonCommand().CobraCommand()
 
 	daemonCommand.PersistentFlags().String(f.Service.Deployer.Environment, "", "Environment name that draughtsman is running in.")
+	daemonCommand.PersistentFlags().String(f.Service.Deployer.Provider, "", "Provider that draughtsman is running in.")
```
Comment (kopiczko) [inline main.go:138]: How is that injected here? I think you'd need to modify the helm chart as well.

Comment (wtty-fool) [reply main.go:138]: I think (after some research) we are using a configmap and secret from installations. They are mounted as config files and then `viper` reads them and parses them into flags.

That's how `projectList` got here before.

Comment (wtty-fool) [reply main.go:138]: See: https://github.com/giantswarm/installations/pull/1250

File: service/eventer/projects.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	}
+	azureProjectList = []string{
+		"azure-app-collection",
+		"azure-operator",
+	}
+	perInstallationProjectLists = map[string][]string{
+		"argali":  []string{"route53-manager"},
+		"axolotl": []string{"route53-manager"},
+		"gauss":   []string{"release-bot"},
+		"giraffe": []string{"desmotes", "route53-manager"},
```
Comment (JosephSalisbury) [inline service/eventer/projects.go:30]:
```suggestion
		"giraffe": []string{"route53-manager"},
```

Comment (JosephSalisbury) [reply service/eventer/projects.go:30]: we don't need `desmotes` anywhere, that project is vvv dead

File: service/eventer/projects.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		"vault-exporter",
+	}
+	awsProjectList = []string{
+		"aws-app-collection",
+		"aws-operator",
+	}
+	kvmProjectList = []string{
+		"kvm-app-collection",
+		"kvm-operator",
+		"ingress-operator",
```
Comment (JosephSalisbury) [inline service/eventer/projects.go:20]: Is this a thing anymore? I think it was merged into `kvm-operator`. https://github.com/giantswarm/ingress-operator is archived :D

**Review by JosephSalisbury — COMMENTED** [review]
Some things we can drop <3

**Comment by kopiczko** [comment]
I see. I left a comment in the other PR.

File: main.go (hunk 1 above)
Comment (kopiczko) [reply main.go:138]: I see. I left a comment in the other PR.

File: service/eventer/projects.go (hunk 3 above)
Comment (wtty-fool) [reply service/eventer/projects.go:20]: Wow, a lot of dead repositories :D

File: main.go (hunk 1 above)
Comment (wtty-fool) [reply main.go:138]: Resolved in the other PR.

File: helm/draughtsman-chart/templates/configmap.yaml [hunk 4]
```diff
@@ -5,6 +5,9 @@ metadata:
   namespace: draughtsman
 data:
   config.yaml: |
+    service:
+      deployer:
+        provider: {{ .Values.Installation.V1.Provider.Kind }}
```
Comment (wtty-fool) [inline helm/draughtsman-chart/templates/configmap.yaml:10]: `service.deployer` section already exists in the secret. It's ok. `viper` is smart enough to merge the secret and configmap into a config. Tested.

Comment (wtty-fool) [reply helm/draughtsman-chart/templates/configmap.yaml:10]: Actually, credit where credit is due, it's `microkit` who's smart about multiple config files https://github.com/giantswarm/microkit/blob/master/flag/flag.go#L41

**Review by kopiczko — APPROVED** [review]

**Pull request merged**